core/pictureobj.cpp

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "pictureobj.h"
+#include "qthelper.h"
+
+PictureObj::PictureObj() : offset({ 0 }), location({ 0 })
+{
+}
+
+PictureObj::PictureObj(const picture &pic) : filename(pic.filename), offset(pic.offset), location(pic.location)
+{
+}
+
+picture PictureObj::toCore() const
+{
+	return picture {
+		copy_qstring(filename),
+		offset,
+		location	
+	};
+}
+
+// Note: this is not strictly correct, as here we sort the filename
+// according to Unicode rules and in the backend a plain byte stream.
+// However, having two pictures with exactly the same timestamp
+// should be rare.
```

Comment:
I thought about this and came up with a simple solution:

We use `std::string` instead of `QString` to keep track of picture filenames. The backing store of an `std::string` is guaranteed to be a zero-terminated C-string, so you can apply all the C-functions on it.

If you argue with consistency, I bring the counter-argument that filenames are *not* usual UI text. On the contrary, it seems rather brittle to convert filenames from system encoding to UTF-16 and then back. Is that even guaranteed to be lossless? With all these unicode combination of characters, etc?

Reply:
That seems perfectly fine. I tend to use `QString` because it's easy, but if using `std::string` makes this easier, more power to you!

Reply:
Brrrr... The interaction of `QString` with `std::string` is annoying (e.g. no implicit conversion to `QString` or `QVariant`). One might get the impression that the Qt project wants to force their data structures down your throat.

Reply:
Yeah, I had that perception before - they want people to stay within the Qt data structure families